lib/android/app/src/reactNative71/java/com/reactnativenavigation/react/ReactGateway.java

```diff
@@ -4,21 +4,22 @@
 import android.content.Intent;
 import android.content.res.Configuration;
 
+import com.facebook.react.ReactHost;
 import com.facebook.react.ReactNativeHost;
 import com.reactnativenavigation.NavigationActivity;
 
 import androidx.annotation.NonNull;
 
 public class ReactGateway {
 
-	private final ReactNativeHost host;
+	private final ReactHost host;
 	private final NavigationReactInitializer initializer;
 	private final JsDevReloadHandler jsDevReloadHandler;
 
-    public ReactGateway(ReactNativeHost host) {
+    public ReactGateway(ReactHost host, boolean isDebug) {
         this.host = host;
-        initializer = new NavigationReactInitializer(host.getReactInstanceManager(), host.getUseDeveloperSupport());
-        jsDevReloadHandler = new JsDevReloadHandler(host.getReactInstanceManager().getDevSupportManager());
+        initializer = new NavigationReactInitializer(host, isDebug);
```

Comment:
Looks like there is not real reason to specify `isDebug` from outside. WDYT about:
```suggestion
    public ReactGateway(ReactApplication reactApplication) {
        this.host = reactApplication.getReactHost();
        initializer = new NavigationReactInitializer(host, reactApplication.getReactNativeHost().getUseDeveloperSupport());
```

Reply:
Or:
```
    public ReactGateway(ReactHost host, ReactNativeHost nativeHost) {
        this.host = host;
        initializer = new NavigationReactInitializer(host, nativeHost.getUseDeveloperSupport());
```

Reply:
Anyways `isDebug` can mean too many things, it's not too clear what's needed